examples/CMakeLists.txt

```diff
@@ -15,6 +15,9 @@ add_executable(non_uniform_heat_equation non_uniform_heat_equation.cpp)
 target_link_libraries(non_uniform_heat_equation PUBLIC DDC::core)
 
 if("${DDC_BUILD_PDI_WRAPPER}")
+    target_compile_definitions(heat_equation PUBLIC "DDC_EXAMPLE_WITH_PDI")
+    target_compile_definitions(uniform_heat_equation PUBLIC "DDC_EXAMPLE_WITH_PDI")
+    target_compile_definitions(non_uniform_heat_equation PUBLIC "DDC_EXAMPLE_WITH_PDI")
```

Comment:
It is not very clear why you introduced this macro. For the `install_test`, you are checking with 
`#if !__has_include(<ddc/pdi.hpp>)`

Reply:
For now, a user/an example cannot rely on `__has_include(<ddc/pdi.hpp>)` to know whether the PDI wrapper is available. The only way to know is to request the related component.

For the test the logic is different, the component is required, so I expect `__has_include(<ddc/pdi.hpp>)` to be true.

Does it make sense ?

Reply:
Sure, now I understand the logic. 
I may prefer a single PDI enabled example like `heat_equation_pdi` without `DDC_EXAMPLE_WITH_PDI `.
This is actually the case for FFT and Splines.

In my opinion, the problem of `DDC_EXAMPLE_WITH_PDI ` is that this may look like a macro available for user to test the availability of PDI.

Reply:
> Sure, now I understand the logic. I may prefer a single PDI enabled example like `heat_equation_pdi` without `DDC_EXAMPLE_WITH_PDI `. This is actually the case for FFT and Splines.

I would like to be able to compile examples without PDI. Duplicating the heat_equation example only to add PDI seems too much to me. 

> In my opinion, the problem of `DDC_EXAMPLE_WITH_PDI ` is that this may look like a macro available for user to test the availability of PDI.

I understand, I thought the `EXAMPLE` was clear enough.

I can also suggest to remove PDI from existing examples and design a new but simpler example dedicated to PDI ? It would not be compiled if PDI is not available.

Reply:
> > Sure, now I understand the logic. I may prefer a single PDI enabled example like `heat_equation_pdi` without `DDC_EXAMPLE_WITH_PDI `. This is actually the case for FFT and Splines.
> 
> I would like to be able to compile examples without PDI. Duplicating the heat_equation example only to add PDI seems too much to me.

I agree with that

> > In my opinion, the problem of `DDC_EXAMPLE_WITH_PDI ` is that this may look like a macro available for user to test the availability of PDI.
> 
> I understand, I thought the `EXAMPLE` was clear enough.
> 
> I can also suggest to remove PDI from existing examples and design a new but simpler example dedicated to PDI ? It would not be compiled if PDI is not available.

If you can do that, it sounds the most reasonable. A DDC version of `pdi-hello-world` is nicer

Reply:
see #691